### PR TITLE
add goals to generate a mapping change diff report

### PR DESCRIPTION
**Pull request by twhetzel**
addresses https://github.com/monarch-initiative/mondo/issues/7230

### CODE REVIEW

File: src/ontology/mondo.Makefile
Comment (matentzn) [inline src/ontology/mondo.Makefile:657]:
```suggestion
tmp/mondo-released.obo: .FORCE
```

I know this is a last resort dependency but we should consider it as this code should always be re-run when the diff is rerun. The SOP takes care of it as well, as it instructs the use of -B, but this is just safer

File: src/ontology/mondo.Makefile
Comment (matentzn) [inline src/ontology/mondo.Makefile:658]:
```suggestion
	wget http://purl.obolibrary.org/obo/mondo/$(KGCL_ONTOLOGY) -O $@
```

So we compare the same things.

**Review by matentzn — CHANGES_REQUESTED** [review]
1 important, 1 minor suggestion

File: src/ontology/mondo.Makefile
Comment (matentzn) [inline src/ontology/mondo.Makefile:647]:
```suggestion
all: kgcl-diff
```

**Review by matentzn — CHANGES_REQUESTED** [review]
Another thing I just saw

**Review by matentzn — APPROVED** [review]
LGTM, assuming you tried this!

**Comment by twhetzel** [comment]
No, actually I have not but I will.

File: src/ontology/mondo.Makefile
Comment (twhetzel) [inline src/ontology/mondo.Makefile:667]: @matentzn how does this get run?

**Review by twhetzel — COMMENTED** [review]
If I run `sh run.sh make kgcl-diff -B` I get:
```
...
runoak -i simpleobo:tmp/mondo-released.obo diff -X simpleobo:mondo-base.obo -o reports/difference_release_base.yaml --output-type yaml
runoak -i simpleobo:tmp/mondo-released.obo diff -X simpleobo:mondo-base.obo \
-o reports/difference_release_base.tsv --output-type csv --statistics --group-by-property oio:hasOBONamespace
runoak -i simpleobo:tmp/mondo-released.obo diff -X simpleobo:mondo-base.obo -o reports/difference_release_base.md --output-type md
No handler for key: RemoveNodeFromSubset
```

When I tried: `sh run.sh make all -B` I ran into this error:
```
MISSING FILE ERROR file '../sparql/preprocess-module.ru' does not exist
```


It runs in "obolibrary/odkfull:latest" and I pulled the latest.

Edit: Each `make` command above resulted in failure with the error messages as posted above.

File: src/ontology/mondo.Makefile
Comment (matentzn) [reply src/ontology/mondo.Makefile:667]: Its not, I removed it!

**Comment by matentzn** [comment]
> No handler for key: RemoveNodeFromSubset

@hrshdhgd?

> MISSING FILE ERROR file '../sparql/preprocess-module.ru' does not exist

I will track that down for the sake of identifying the leak, but in general we should never run:

```
sh run.sh make all -B
```

Always run:

```
sh run.sh make all -B IMP=false
```

Somewhere in the main pipeline there is an import dependency that should not exist anymore, but this is unrelated to this PR.

**Comment by hrshdhgd** [comment]
> > No handler for key: RemoveNodeFromSubset
> 
> @hrshdhgd?

Sorry about that. Somehow this got missed in `oaklib`. Working a PR right now.

**Comment by twhetzel** [comment]
I ran `sh run.sh make all -B IMP=false` and the release process finished successfully. 

I then checked `reports/difference_release_base.md` against the most recent [Orphanet alignment for August release](https://github.com/monarch-initiative/mondo/pull/7933) and [OMIM Alignment for August release](https://github.com/monarch-initiative/mondo/pull/7932) and the new xrefs added into Mondo are found in the new report, e.g. `reports/difference_release_base.md`.

**Comment by twhetzel** [comment]
@matentzn anything else that should be checked or done before merging?

**Comment by matentzn** [comment]
No, good to go!

**Comment by hrshdhgd** [comment]
> > > No handler for key: RemoveNodeFromSubset
> > 
> > 
> > @hrshdhgd?
> 
> Sorry about that. Somehow this got missed in `oaklib`. Working on a [PR](https://github.com/INCATools/ontology-access-kit/pull/783) right now.

`oaklib` v0.6.12 should take care of this

**Comment by matentzn** [comment]
Thank you @hrshdhgd

It will be a while before we will notice this change, we just updated ODK and it will be at least until October now :P thanks for fixing! @twhetzel i think we can merge this!

**Pull request merged**
